docs/Coding_Conventions_and_Standards.md

```diff
@@ -23,6 +23,9 @@ Other
 * When adding a new class, disable copy/assignment/move until you have a proven need for these capabilities. If a need arises, enable copy/assignment/move selectively, and when doing so validate that the implementation of the class supports what is being enabled.
   * Use ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE initially
   * See the other ORT_DISALLOW_* macros in https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/common/common.h
+* Consider using const gsl::span<const T>& (or std::span when supported) as input arguments when passing const references to vector like containers. This allows to make the function container independent, represent arbitrary memory spans or pass sub-spans into it.
+* Use TensorShapeVector to build or modify shapes from core/framework/tensor_shape.h. It is based on a vector implementation that features small buffer optimization. Its small buffer size is the same that we use in TensorShape. Use InlinedShapeVectorT<T> for shape related operations, but of different type.
+* Use InlinedHashSet<T> and InlinedHashMap<T> from core/framework/inlined_containers.h. These are drop-in replacements for std::unordered_set/map that store their keys and values in one continuous buffer and reduce the number of frequency allocations. They also do not allocate an 'end' node. Note, that these Hash containers do not provide pointer stability. Use InlinedVector<T, N> when there is a reason to believe your code may benefit from small buffer optimization and you have a good idea (data) what size it must be.
```

Comment:
```suggestion
* Use InlinedHashSet<T> and InlinedHashMap<T> from core/framework/inlined_containers.h. These are drop-in replacements for std::unordered_set/map that store their keys and values in one continuous buffer and reduce the number of frequent per-node allocations. They also do not allocate an 'end' node. Note, that these Hash containers do not provide pointer stability. Use InlinedVector<T, N> when there is a reason to believe your code may benefit from small buffer optimization and you have a good idea (data) what size it must be.
```

Can also point to the https://abseil.io/docs/cpp/guides/container if someone wants to read more.

Reply:
Done.